src/backend/metal/src/device.rs

```diff
@@ -1001,14 +1015,20 @@ impl hal::Device<Backend> for Device {
             let allocations = memory.allocations.lock().unwrap();
             for (alloc_range, dst) in allocations.find(&range) {
                 let start = alloc_range.start.max(range.start);
-                let end = alloc_range.end.min(range.end);
+                let size = alloc_range.end.min(range.end) - start;
+                let other = start - alloc_range.start;
+                assert_eq!(other & ALIGNM_MASK, 0);
+                if size & ALIGNM_MASK != 0 || start & ALIGNM_MASK != 0 {
+                    warn!("Invalidation of {:?} doesn't respect `non_coherent_atom_size`",
+                        start .. start + size);
+                }
                 command::exec_blit(encoder, &soft::BlitCommand::CopyBuffer {
                     src: cpu_buffer.clone(),
                     dst,
                     region: BufferCopy {
-                        src: start,
-                        dst: start - alloc_range.start,
-                        size: end - start,
+                        src: start & !ALIGNM_MASK,
+                        dst: other,
+                        size: size & !ALIGNM_MASK,
```

Comment:
Minor but should probably prefer `end & !ALIGNM_MASK - start & !ALIGNM_MASK` here (and below) instead